Geometry/TrackerGeometryBuilder/src/trackerHierarchy.cc

```diff
@@ -1,39 +1,32 @@
-#include "DataFormats/SiPixelDetId/interface/PXBDetId.h"
-#include "DataFormats/SiPixelDetId/interface/PXFDetId.h"
-#include "DataFormats/SiStripDetId/interface/TIBDetId.h"
-#include "DataFormats/SiStripDetId/interface/TIDDetId.h"
-#include "DataFormats/SiStripDetId/interface/TOBDetId.h"
-#include "DataFormats/SiStripDetId/interface/TECDetId.h"
+#include "Geometry/TrackerGeometryBuilder/interface/trackerHierarchy.h"
 
 #include<string>
 #include<vector>
 
-std::string trackerHierarchy(unsigned int rawid) {
+std::string trackerHierarchy(const TrackerTopology* tTopo, unsigned int rawid) {
   DetId id(rawid);
   int subdetid = id.subdetId();
   switch (subdetid) {
     
     // PXB
   case 1:
     {
-      PXBDetId module(rawid);
-      char theLayer  = module.layer();
-      char theLadder = module.ladder();
-      char theModule = module.module();
+      char theLayer  = tTopo->layer(id);
```

Comment:
since we have the explicit checks, we can write pxbLayer, pxbModule, and (below) pxfModule etc. - I don't feel strongly about this, but it's a bit more clear